File: Nemes Tihamér/Days of past/Hétfő/1. Fadöntés (moho, mester)/Fadontes.cs
using System;

class Fadontes
{
    public const int MAXN = 1000;

    public class Fa
    {
        public int tav, mag;
        public Fa(int tav, int mag)
        {
            this.tav = tav;
            this.mag = mag;
        }
    }

    public static void Main()
    {
        int n;
        n = Convert.ToInt32(Console.ReadLine());
        
        Fa[] adatok = new Fa[MAXN];

        for(int i = 0; i < n; i++)
        {
            String[] temp = Console.ReadLine().Split();
            int tav = Convert.ToInt32(temp[0]);
            int mag = Convert.ToInt32(temp[1]);
            adatok[i] = new Fa(tav, mag);
        }

        int db = 1; //kidöntendő fák száma, elsőt mindenképp ki kell
        int lastmaxindex = 0; //legtovább elérő kidöntött fa indexe
        int maxi = 0, maxe = 0; //a legtöbbet kidöntő fa indexe és maximum értéke
        int loki = 0, lokdb = 0; //a jelenleg vizsgált fa hányat dönt ki: addig növeljük míg dönti ki a fákat

        for (int i = 1; i < n; i++)
        {
            int tav = adatok[i].tav - adatok[lastmaxindex].tav;
            //ha nem dől ki
            if(adatok[lastmaxindex].mag <= tav)
            {
                db++;
                if(lokdb > maxe)
                {
                    maxe = lokdb;
                    maxi = loki;
                }
                loki = i;
                lokdb = 0;
            }
            else
            {
                lokdb++;
            }
            
            //ha a mostani tovább elér
            if(adatok[i].mag > adatok[lastmaxindex].mag - tav)
            {
                lastmaxindex = i;
            }
        }
        //utolsó fa is döntheti ki a legtöbbet
        if(lokdb > maxe)
        {
            maxe = lokdb;
            maxi = loki;
        }

        Console.WriteLine(db);
        Console.WriteLine(maxi + 1);

    }
}

File: Nemes Tihamér/Topics/6. GREEDY/Haladó/Fadöntés (8. moho, mester)/fadontes.pas
Program fadontes;

const MAXN = 1000;

Type fa = Record
    tav, mag: integer;
	end;

Var
	n: integer;
	adatok: array[0..MAXN-1] of fa;
	db, lastmaxindex, maxi, maxe, loki, lokdb, tav, i: integer;

Begin

    Readln(n);

    for i := 0 to n-1 do
	Begin
		Readln(adatok[i].tav, adatok[i].mag);
	End;

    db := 1; //kidöntendő fák száma, elsőt mindenképp ki kell
    lastmaxindex := 0; //legtovább elérő kidöntött fa indexe
    maxi := 0; maxe := 0; //a legtöbbet kidöntő fa indexe és maximum értéke
    loki := 0; lokdb := 0; //a jelenleg vizsgált fa hányat dönt ki: addig növeljük míg dönti ki a fákat

	for i := 1 to n-1 do
	Begin
		tav := adatok[i].tav - adatok[lastmaxindex].tav;
		//ha nem dől ki
        if(adatok[lastmaxindex].mag <= tav) then
		Begin
			inc(db);
			if(lokdb > maxe) then
			Begin
				maxe := lokdb;
				maxi := loki;
			End;
			loki := i;
			lokdb := 0;
			
		End
        else
        Begin
			inc(lokdb);
		End;
        
        //ha a mostani tovább elér
        if(adatok[i].mag > adatok[lastmaxindex].mag - tav) then
        Begin
            lastmaxindex := i;
        End;
	End;

    //utolsó fa is döntheti ki a legtöbbet
    if(lokdb > maxe) then
    Begin
        maxe := lokdb;
        maxi := loki;
    End;

	Writeln(db);
	Writeln(maxi + 1);

End.